### PR TITLE
Add AuthenticationStrategy, ManagerDn, ManagerPassword, IdentityStrat…

### LINKED ISSUE #65: LDAP integration: no option to provide properties “manager dn”, “manager password”, “authentication strategy”.
From nifikop created by [ayush7007](https://github.com/ayush7007): Orange-OpenSource/nifikop#180  

I am integrating NIFI with ldap however can only define below option ( available under "LdapConfiguration" )- 

image

This would need other properties as well to integrate successfully such as “manager dn”, “manager password”, “authentication strategy”, etc.
image

I don’t see any option available to provide these Ldap properties. Also overriding “login-identity-providers.xml” is not available.
Please let me know the way to provide these values, or whether it's available ?

### PR DESCRIPTION
**Pull request by pashtet04**
…egy properties for LDAP integration

| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | yes
| Deprecations?   | no
| Related tickets | fixes #65 
| License         | Apache 2.0

### What's in this PR?
Add AuthenticationStrategy, ManagerDn, ManagerPassword, IdentityStrategy properties for LDAP integration

### Why?
Because LDAP can be secured 

### Checklist

- [x ] Implementation tested
- [ ] User guide and development docs updated (if needed)
- [x] Append changelog with changes

### To Do

- [ ] Add properties for TLS

### CODE REVIEW

File: pkg/resources/templates/config/login_identity_providers.go [hunk 1]
```diff
@@ -68,9 +68,9 @@ var LoginIdentityProvidersTemplate = `<?xml version="1.0" encoding="UTF-8" stand
     <provider>
         <identifier>ldap-provider</identifier>
         <class>org.apache.nifi.ldap.LdapProvider</class>
-        <property name="Authentication Strategy">START_TLS</property>
```
Comment (mh013370) [inline pkg/resources/templates/config/login_identity_providers.go:71]: Introducing these configurations without defaults could break users who deployed this previously and benefitted from "Authentication Stategy"="START_TLS" and "Identity Strategy"="USE_DN". I'll recommend some changes

File: pkg/resources/templates/config/login_identity_providers.go [hunk 2]
```diff
@@ -68,9 +68,9 @@ var LoginIdentityProvidersTemplate = `<?xml version="1.0" encoding="UTF-8" stand
...
-        <property name="Manager DN"></property>
-        <property name="Manager Password"></property>
+        <property name="Authentication Strategy">{{.LdapConfiguration.AuthenticationStrategy}}</property>
```
Comment (mh013370) [inline pkg/resources/templates/config/login_identity_providers.go:71]:
```suggestion
        <property name="Authentication Strategy">{{.LdapConfiguration.AuthenticationStrategy}}</property>
        <property name="Authentication Strategy">{{or .LdapConfiguration.AuthenticationStrategy "START_TLS"}}</property>
```

File: pkg/resources/templates/config/login_identity_providers.go [hunk 3]
```diff
@@ -87,7 +87,7 @@ var LoginIdentityProvidersTemplate = `<?xml version="1.0" encoding="UTF-8" stand
         <property name="Url">{{.LdapConfiguration.Url}}</property>
         <property name="User Search Base">{{.LdapConfiguration.SearchBase}}</property>
         <property name="User Search Filter">{{.LdapConfiguration.SearchFilter}}</property>
-        <property name="Identity Strategy">USE_DN</property>
+        <property name="Identity Strategy">{{.LdapConfiguration.IdentityStrategy}}</property>
```
Comment (mh013370) [inline pkg/resources/templates/config/login_identity_providers.go:90]:
```suggestion
        <property name="Identity Strategy">{{.LdapConfiguration.IdentityStrategy}}</property>
        <property name="Identity Strategy">{{or .LdapConfiguration.IdentityStrategy "USE_DN"}}</property>
```

File: api/v1alpha1/nificluster_types.go [hunk 4]
```diff
@@ -495,6 +495,18 @@ type LdapConfiguration struct {
 	// Filter for searching for users against the 'User Search Base'.
 	// (i.e. sAMAccountName={0}). The user specified name is inserted into '{0}'.
 	SearchFilter string `json:"searchFilter,omitempty"`
+	// How the connection to the LDAP server is authenticated.
```
Comment (mh013370) [inline api/v1alpha1/nificluster_types.go:498]: @juldrixx : Are we adding new API changes to only `v1` or to both `v1` and `v1alpha1`?

Comment (pashtet04) [reply api/v1alpha1/nificluster_types.go:498]: Should I remove it from `v1alpha` ?

Comment (pashtet04) [reply api/v1alpha1/nificluster_types.go]: [#02b](https://github.com/konpyutaika/nifikop/pull/266/commits/02bdf77bd5ca544a34b74fb543950c48b4befc4d)

**Comment by mh013370** [comment]
Can you please update the changelog with the changes?

**Review by mh013370 — APPROVED** [review]

**Comment by mh013370** [comment]
Ah apologies - all commits need to be signed 😄 Then i can merge.

**Comment by pashtet04** [comment]
Verified

**Comment by mh013370** [comment]
Thank you for the contribution! 😄

**Pull request merged**